perfkitbenchmarker/linux_benchmarks/object_storage_service_benchmark.py

```diff
@@ -402,42 +401,38 @@ def _ProcessMultiStreamResults(raw_result, operation, sizes,
   distribution_metadata = metadata.copy()
   distribution_metadata['object_size_B'] = 'distribution'
 
-  latency_prefix = 'Multi-stream %s latency ' % operation
+  latency_prefix = 'Multi-stream %s latency' % operation
   logging.info('Processing %s multi-stream %s results for the full '
                'distribution.', len(records_in_interval), operation)
-  analyze.AppendStatsAsSamples(
+  _AppendPercentilesToResults(
+      results,
       records_in_interval['latency'],
+      latency_prefix,
       'sec',
-      results,
-      name_prefix=latency_prefix,
-      metadata=distribution_metadata)
+      distribution_metadata)
 
   logging.info('Processing %s multi-stream %s results for net throughput',
                len(records), operation)
-  net_thpt, thpt_with_overhead, total_overhead, overhead_prop = (
-      analyze.AllStreamsThroughputStats(
-          records_in_interval['latency'],
-          records_in_interval['size'],
-          records_in_interval['stream_num'],
-          num_streams,
-          all_streams_duration))
+  throughput_stats = analyze.ThroughputStats(
+      records_in_interval['start_time'],
+      records_in_interval['latency'],
+      records_in_interval['size'],
+      records_in_interval['stream_num'],
+      num_streams)
+  gap_stats = analyze.GapStats(
+      records['start_time'],
+      records['latency'],
+      records['stream_num'],
+      all_streams_active,
+      num_streams)
   logging.info('Benchmark overhead was %s percent of total benchmark time',
-               overhead_prop * 100.0)
-
-  results.extend([
-      sample.Sample(
-          'Multi-stream %s net throughput' % operation,
-          net_thpt, 'B/sec', metadata=distribution_metadata),
-      sample.Sample(
-          'Multi-stream %s net throughput (experimental)' % operation,
-          thpt_with_overhead, 'B/sec', metadata=distribution_metadata),
-      sample.Sample(
-          'Multi-stream %s total overhead time' % operation,
-          total_overhead, 'sec', metadata=distribution_metadata),
-      sample.Sample(
-          'Multi-stream %s overhead time proportion' % operation,
-          overhead_prop * 100.0, 'percent', metadata=distribution_metadata)])
+               gap_stats['gap time proportion'].magnitude)
 
+  for name, value in itertools.chain(throughput_stats.iteritems(),
```

Comment:
As discussed in person, we should add a very simplified version of aggregated throughput computed in following way:

(Total bytes transferred by any streams in this run) / total time taken
(my understanding is the any_streams_active array could be used for this purpose)

We can name this as "net throughput (simplified)"

We will see how these 3 numbers diverge and under which cases.